### ISSUE TITLE
[Feature]: Rewrite share sheet UI using SwiftUI

### ISSUE THREAD

**Issue by tmolitor-stud-tu** [monal-im/Monal#1116]
### Precondition

- [X] I understood that this template is only for feature requests and not for bug reports

### Cross-checked related issues?

- [X] I have cross-checked this overview https://github.com/monal-im/Monal/issues/322 as well as filtered for related labels https://github.com/monal-im/Monal/labels

### Describe your feature

The title says it all: replace the old and ugly share sheet UI with a modern an polished SwiftUI version but provide the same functionality.

**Comment by NafSyed** [comment]
Hey, I would like to work on this with you!!

**Comment by NafSyed** [comment]
I have done some minor bug fixes to DebugView, AddContactMenu, PasswordMigration but couldn't commit so, kindly allow me access to commit the changes

**Comment by thevaidik** [comment]
You can't directly commit and push in main repo. Fork it and raise a pull request.

**Comment by tmolitor-stud-tu** [comment]
> Hey, I would like to work on this with you!!

No, please improve your skills significantly and come back afterwards.
